cmd/jujud/agent/machine/manifolds.go

```diff
@@ -289,6 +292,11 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 		return crosscontroller.NewClient(conn), nil
 	}
 
+	var externalUpdateProxyFunc func(proxy.Settings) error
+	if runtime.GOOS == "linux" && !config.IsCaasConfig {
```

Comment:
Is runtime.GOOS going to cause issues in tests? Normally we use os.HostOS()

Reply:
There's a few other places where we use this instead. Using os.HostOS() seems to be for those times we want to check for == Windows. Here, we want linux (of any variant) and our os.HostOS() has many different linux variants, so we'd need a switch of something and then it gets fragile.